File: Sem1/ProjetoIntegrador/exercicios/src/tarefaIII/L03AlgoritimosPg42/L03A.java
package tarefaIII.L03AlgoritimosPg42;

import javax.swing.*;

/*
a. Ler dois valores numéricos inteiros e
apresentar o resultado da diferença do maior pelo menor valor.
*/
public class L03A {
    static void main() {
        double value1 = 0, value2 = 0, total;
        boolean validation = true;

        while (validation) {
            try {
                value1 = Double.valueOf(JOptionPane.showInputDialog("\nDigite o valor do primeiro termo: ").replace(",", "."));
                while (validation) {
                    try {
                        value2 = Double.valueOf(JOptionPane.showInputDialog("\nDigite o valor do segundo termo: ").replace(",", "."));
                        validation = false;
                    } catch (NumberFormatException e) {
                        JOptionPane.showMessageDialog(null, "\nValor do segundo termo inválido!, digite novamente");
                    }
                }
            } catch (NumberFormatException e) {
                JOptionPane.showMessageDialog(null, "\nValor do primeiro termo inválido!, digite novamente");
            }

        }
        total = highestValue(value1, value2);
        JOptionPane.showMessageDialog(null, "\n\nA diferença entre " + String.format("%.2f", value1)
                + " e " + String.format("%.2f", value2)+" é " + String.format("%.2f", total));
    }

    public static double highestValue(double valor1, double valor2) {
        double result;
        if (valor1 > valor2) {
            result = valor1 - valor2;
        } else {
            result = valor2 - valor1;
        }
        return result;
    }
}

File: Sem1/ProjetoIntegrador/exerciciosPyton/L03 - algoritimos - pg42/L03A.py
#-*-coding:utf-8-*-
#a. Ler dois valores numéricos inteiros e 
#apresentar o resultado da diferença do maior pelo menor valor. 
value1 = 0
value2 = 0

def highestValue (value1, value2):
    if(value1 > value2):
        total = value1 - value2
    else:
        total = value2 - value1
        return total

validation = True    

while (validation):
    try:
        value1 = float(input("\nDigite o valor do primeiro termo: ").replace(",","."))
        while (validation):
            try:
                value2 = float(input("\nDigite o valor do segundo termo: ").replace(",","."))
                validation =  False
            except:
                    print("\nValor do segundo termo inválido!, digite novamente")
    except:
        print("\nValor do primeiro termo inválido!, digite novamente")
total = highestValue(value1,value2)
print(f"\n\nA diferença entre {value1:.0f} e {value2:.0f} é {total:.0f}")